### PR TITLE
Initial recipe for arping-2.16

### PR DESCRIPTION
**Pull request by jimklimov**
How could we not have this? ;)

### CODE REVIEW

**Comment by pyhalov** [comment]
Should it work in a zone with exclusive IP stack?  

pfexec arping GW
ARPING GW
Timeout
Timeout
Timeout

File: components/sysutils/arping/Makefile [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+# Note: A few options below were replicated from tcpdump and fping recipes
+# fix my alignment issues instead of falling over on sparc
+studio_ALIGN.sparc.64=-xmemalign=16i
+CONFIGURE_OPTIONS.sparc += ac_cv___attribute__=no
+CONFIGURE_OPTIONS += $(CONFIGURE_OPTIONS.$(MACH))
+# lazyloading (indirectly enabled by -Bdirect) seems to cause problems with
+# libpcap device enumeration.
+LD_B_DIRECT=
+ASLR_MODE =	$(ASLR_ENABLE)
```
Comment (xen0l) [inline components/sysutils/arping/Makefile]: Remove this, we don't have ASLR capabilities on illumos

**Comment by jimklimov** [comment]
It should, and sometimes works for me - though usually only for one
"arping" packet in a long while, e.g. usually the first one and then for a
minute none others pass. I am not sure where in the stack these get lost :(

2016-05-24 23:29 GMT+02:00 Alexander Pyhalov notifications@github.com:

**Comment by pyhalov** [comment]
this issue should be investigated before merge

**Comment by jimklimov** [comment]
From reading similar code (snoop, p0f, nmap, etc.) it seems that some ioctl's are generally in order to force Solarish systems to not-buffer incoming packets. Otherwise you have to wait a while before you see some output from the sniffer.

I see that the main ioctl is there in arping, and others I tried to stuff in there now (to set size and timeout limits for the select() waits to 0) fail with Invalid Parameter (22).

**Comment by jimklimov** [comment]
I did some more research this weekend regarding issues with arping not seeing all traffic - same applies to p0f, and seems to be some change in illumos vs. sol10 where ioctl's to change the STREAMS attributes for the pcap socket do work, and on OI/Hipster they return EINVAL.

The ability to quickly poll incoming packets by a sniffer is there, since snoop (libdlpi) and tcpdump (libpcap) work. "Surprisingly" tcpdump does work on OI, and with same libpcap as does not work for arping/p0f, so one has to dig into code to find differences (who uses which bpf implementation, what ioctl's are used and do work, what flags are used to set up pcap, notably what timeouts are in place for pcap live stream setup and for poll or select - and whether poll or select are used, etc.)

Manpages of libpcap mention that some OSes are worse at poll() of pcap sockets than others (but suggested workarounds did not help me)

**Comment by jimklimov** [comment]
The sniffers apparently need some more debugging on illumos or specifically OI, or maybe just source code inspection  - they are not getting packets, or not many/often, which is seemingly similar to issues with (disabling of) buffering of accumulated data. There are other sniffers in repo that work (nmap, ngrep, tcpdump come to mind first). All are based on libpcap as well.

So there is a need for some quiet time to sit down and compare who does what - and I didn't find enough of that limited resource in months ;(

**Comment by xen0l** [comment]
I am closing this PR as there hasn't been any activity lately.

**Pull request closed without merging**